Homework/homework10/task.py

```diff
@@ -0,0 +1,150 @@
+import asyncio
+import heapq
+import json
+import re
+from datetime import datetime
+from typing import Dict, List, Tuple
+
+import aiohttp
+
+from bs4 import BeautifulSoup
+
+
+def get_company_name(company_page_soup) -> str:
+    """Get company name from the page."""
+    return company_page_soup.find(
+        "span", {"class": "price-section__label"}
+    ).text.strip()
+
+
+def get_company_code(company_page_soup) -> str:
+    """Get company name from the page."""
+    return (
+        company_page_soup.find("div", {"class": "price-section__row"})
+        .find("span", {"class": "price-section__category"})
+        .find("span")
+        .text[2:]
+    )
+
+
+def get_stock_price(company_page_soup, currency: float) -> float:
+    """Get stock price in usd from the page."""
+    stock_price_usd = company_page_soup.find(
+        "span", {"class": "price-section__current-value"}
+    ).text
+    return round(float(stock_price_usd.replace(",", "")) * currency, 2)
+
+
+def get_pe_ratio(company_page_soup) -> float:
+    """Get P/E ratio from the page."""
+    try:
+        return float(
+            company_page_soup.find(
+                "div", string=re.compile("P/E Ratio")
+            ).previous_sibling.replace(",", "")
+        )
+    except AttributeError:
+        return 0
+
+
+def get_low_high(company_page_soup) -> Tuple[float, float]:
+    """Get 52 Week Low and 52 Week High values from the page."""
+    script = company_page_soup.find("div", {"class": "snapshot"}).find("script")
+    low = float(re.search(r"low52weeks: (\d*\.\d+|\d+),", script.string).group(1))
+    high = float(re.search(r"high52weeks: (\d*\.\d+|\d+),", script.string).group(1))
+    return low, high
+
+
+def get_profit(low_high_values: Tuple[float, float]) -> float:
+    """Get a value of profit if we buy the stock at 52 Week Low price and sell at 52 Week High price."""
+    low, high = low_high_values
+    return round((high - low) / low * 100, 2)
+
+
+def sort_by_key(data: List[Dict], value: str) -> List[Dict]:
+    """Sort the company info by given key value."""
+    return heapq.nlargest(10, data, key=lambda x: x[value])
+
+
+async def get_current_currency() -> float:
+    """Get today currency from the bank page."""
+    current_datetime = datetime.now()
+    year = current_datetime.year
+    month = current_datetime.month
+    formatted_month = month if len(str(month)) > 1 else "0" + str(month)
+    day = current_datetime.day
+    formatted_day = day if len(str(day)) > 1 else "0" + str(day)
+    bank_url = f"http://www.cbr.ru/scripts/XML_daily.asp?date_req={formatted_day}/{formatted_month}/{year}"
+    currency_page_soup = BeautifulSoup(await fetch_html(bank_url), "lxml")
+    currency = (
+        currency_page_soup.find("valute", {"id": "R01235"})
+        .find("value")
+        .text.replace(",", ".")
+    )
+    return float(currency)
```

Comment:
Should we round it too?

Reply:
But we don't need to show it anywhere, its used only for calculating stock price which we rounds